plugins/nextcloud/NextcloudAddressBook.php

```diff
@@ -0,0 +1,96 @@
+<?php
+use RainLoop\Providers\AddressBook\Classes\Contact;
+use Sabre\VObject\Component\VCard;
+
+class NextcloudAddressBook implements \RainLoop\Providers\AddressBook\AddressBookInterface
+{
+	use Rainloop\Providers\AddressBook\CardDAV;
+	private const URI = 'webmail';
+	private $contactsManager;
+	private $key;
+
+	function __construct()
+	{
+		$this->contactsManager = \OC::$server->getContactsManager();
+		foreach ($this->contactsManager->getUserAddressBooks() as $addressbook) {
+			if ($addressbook->getUri() !== self::URI) {
+				$this->contactsManager->unregisterAddressBook($addressbook);
+			}
+			else {
+				$this->key = $addressbook->getKey();
+			}
+		}
+	}
+
+	public function IsSupported() : bool {
+		// Maybe just return true, contacts app is just a frontend
```

Comment:
We could maybe support this only if the contacts app is enabled in nextcloud..

but again that is just a frontend, contacts are supported by default in nextcloud via carddav